pkg/tsdb/azuremonitor/utils/utils.go

```diff
@@ -25,7 +27,7 @@ func GetFirstSubscriptionOrDefault(ctx context.Context, dsInfo types.DatasourceI
 
 	res, err := dsInfo.Services["Azure Monitor"].HTTPClient.Do(request)
 	if err != nil {
-		return "", fmt.Errorf("failed to retrieve subscriptions: %v", err)
+		return "", errorsource.DownstreamError(fmt.Errorf("failed to retrieve subscriptions: %v", err), false)
```

Comment:
Why don't we use your new function here `ApplySourceFromError` ?

Reply:
I created that helper to avoid having to do the casting to retrieve the error source. We don't need to do that here because we can set the error source directly on the error.